kotlin/src/main/kotlin/io/github/xanthic/cache/ktx/CacheApiSpecExtensions.kt

```diff
@@ -21,7 +21,7 @@ fun <K, V> processSpec(init: CacheApiSpec<K, V>.() -> Unit): CacheApiSpec<K, V>
 /**
  * @see io.github.xanthic.cache.api.ICacheSpec.provider
  */
-var <K, V> CacheApiSpec<K, V>.provider: CacheProvider?
+var <K, V> CacheApiSpec<K, V>.provider: CacheProvider
```

Comment:
my only concern here is: within the spec consumer of `CacheApiSpec.process`, the provider is temporarily null: https://github.com/Xanthic/cache-api/blob/chore/provider-notnull/core/src/main/java/io/github/xanthic/cache/core/CacheApiSpec.java#L81

we could set it to `defaultCacheProvider` immediately, but it'd be trickier to do the log message that states the default is being used (in a manner that doesn't trigger the message is the provider is explicitly set from the consumer)